### ISSUE TITLE
.then should not be used as it has special behavior in js

### ISSUE THREAD

**Issue by jessekelly881** [gcanti/fp-ts-routing#69]
## 🐛 Bug report

fp-ts-routing should not use .then as this has special behavior within js. Its use is also flagged by eslint. 

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/7d804e3ce8ecb9d51b16d558eeae6dc12a81ec4c/docs/rules/no-thenable.md

* Not specific to a particular version of the api or environment
* This ticket follows this conversation https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1942

**Comment by StefanoMagrassi** [comment]
I don't think that a specific configuration of ESLint made and thought out by a specific person should be marked as "the right one".

`.then` has a "special" behavior only for objects that implement a `Thenable` interface, like `Promise`

TS already gives you an error if you tray to use `fp-ts-routing` `.then` with `await`.

I think that this kind of changes should be considered only in a more generic discussion about releasing a 1.0.0 version of the package.

**Comment by AumyF** [comment]
Exporting a function named `then` makes the module namespace object (result of `import("fp-ts-routing")`) thenable and `await`ing the object lead the runtime to hang or crash.

```ts
// es modules, Node.js v22.10.0

const a = await import("fp-ts-routing");

console.log(typeof a); // not called
```

```shell
❯ node .
Warning: Detected unsettled top-level await at file:///Users/user/test/index.js:1
const a = await import("fp-ts-routing");
          ^
```

For example, Vitest rewrites all static imports to dynamic import and gets stuck because of the issue.  [The maintainers of Vitest also says the issue should be fixed by the package side](https://github.com/vitest-dev/vitest/issues/5122#issuecomment-2406488169).

**Comment by StefanoMagrassi** [comment]
> Warning: Do not export a function called then() from a module. This will cause the module to behave differently when imported dynamically than when imported statically.

🙄 

anyway, the issue here is that renaming `then` would be an actual breaking change.

to me could be ok applying it (choosing a _non-sensitive_ name like, IDK, `andThen`) without forcing to release the v1.0, but we have to make it very clear in order to make anyone aware of this.

@gcanti what do you think? at the end of the day you'll be the one in charge of publishing the package on NPM
